File: BE/Database/Model/TournamentProgress.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Database.Model
{
    public class TournamentProgress
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [ForeignKey("TournamentRegistration")]
        public int TournamentRegistrationId { get; set; }
        public TournamentRegistration TournamentRegistration { get; set; }

        public int RoundNumber { get; set; }
        public bool IsEliminated { get; set; } = false; // True nếu đã bị loại
        public int Wins { get; set; } = 0;
        public int Losses { get; set; } = 0;

        public DateTime LastUpdated { get; set; } = DateTime.UtcNow;
    }
}

File: MB/lib/core/models/tournament_progress.dart
class TournamentProgress {
  final int id;
  final int tournamentRegistrationId;
  final int roundNumber;
  final bool isEliminated;
  final int wins;
  final int losses;
  final DateTime lastUpdated;

  TournamentProgress({
    required this.id,
    required this.tournamentRegistrationId,
    required this.roundNumber,
    required this.isEliminated,
    required this.wins,
    required this.losses,
    required this.lastUpdated,
  });

  factory TournamentProgress.fromJson(Map<String, dynamic> json) {
    return TournamentProgress(
      id: json['id'] as int,
      tournamentRegistrationId: json['tournamentRegistrationId'] as int,
      roundNumber: json['roundNumber'] as int,
      isEliminated: json['isEliminated'] as bool,
      wins: json['wins'] as int,
      losses: json['losses'] as int,
      lastUpdated: DateTime.parse(json['lastUpdated'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'tournamentRegistrationId': tournamentRegistrationId,
      'roundNumber': roundNumber,
      'isEliminated': isEliminated,
      'wins': wins,
      'losses': losses,
      'lastUpdated': lastUpdated.toIso8601String(),
    };
  }
}
